src/NHibernate/Persister/Entity/AbstractEntityPersister.cs

```diff
@@ -4491,5 +4517,48 @@ public string GetInfoString()
 			return MessageHelper.InfoString(this);
 		}
 		#endregion
+
+		internal SqlString GenerateSequentialSelect(AbstractEntityPersister subclassPersister)
+		{
+			//figure out which tables need to be fetched (only those that contains at least a no-lazy-property)
+			var tableNumbers = new HashSet<int>();
+			var props = subclassPersister.PropertyNames;
+			var classes = subclassPersister.PropertySubclassNames;
+			for (var i = 0; i < props.Length; i++)
+			{
+				var propTableNumber = GetSubclassPropertyTableNumber(props[i], classes[i]);
+				if (IsSubclassTableSequentialSelect(propTableNumber) && !IsSubclassTableLazy(propTableNumber))
+				{
+					tableNumbers.Add(propTableNumber);
+				}
+			}
+			if (tableNumbers.Count == 0)
+				return null;
+
+			//figure out which columns are needed (excludes lazy-properties)
+			var columnNumbers = new List<int>();
```

Comment:
The comment lets think some filter on `subclassColumnLazyClosure` is done by the code, but it is not the case. Are the lazy columns really excluded here? I do not think they are.

But it seems to be an old issue of the code, just moved "as-is" from `SingleTableEntityPersister`. If there is something to fix here, better do it in another PR. (Excepted if the fix is just changing the comment.)

There is the same issue for formulas below.

Reply:
> Are the lazy columns really excluded here?

No, they are not. When the filter will be added the #1922 would need to be enhanced to support it as we can [fetch a lazy property of a subclass](https://github.com/nhibernate/nhibernate-core/pull/1922/files#diff-03963f8bf0b4f656910fabd85135c7bcR678). In addition the current `fetch all properties` would also need to be enhanced to support it.